docs/01_cva6_user/CVX_Interface_Coprocessor.rst

```diff
@@ -198,8 +198,183 @@ Never let the CV-X-IF interface unconnected with the ``CVA6ConfigCvxifEn`` param
 
 How to design a coprocessor for the CV-X-IF interface
 -----------------------------------------------------
-*The team is looking for a contributor to write this section.*
+
+We can add a custom coprocessor that implements custom instructions by modifying the example coprocessor in this repository.
+This section is structured as a tutorial to implement two instructions that manipulate binary-coded decimal numbers.
+That is, numbers where each 4-bit nibble represents a single base-10 digit with the value 0-9.
+For example, 123 in decimal = 0x7B in hexadecimal = 0x123 in binary-coded decimal.
+
+#. Specify your new instructions
+
+    The example coprocessor defines instructions for both the custom 0 and custom 1 major opcodes.  Using a standard R-type format, each of these allows 1024 distinct instructions to be defined using the 7-bit funct7 field and the 3-bit funct3 field.
+
+    .. image:: rtype_format.png
+      :width: 400
+      :alt: Rtype RISC-V instruction format
+
+    Example:
+
+    .. code-block::
+
+      opcode=custom1, funct7=0x00, funct3=0x00: BCDfromBin
+      rf[rd] <- BCD(rf[rs1])
+      Register rd is written with the binary-coded decimal equivalent of the binary integer value in rs1.
+      Note: rs2 is not used.
+
+      opcode=custom1, funct7=0x00, funct3=0x01: BCDADD
+      rf[rd] <- ADD.BCD(rf[rs1], rf[rs2])
+      Register rd is written with a binary-coded decimal (BCD) sum of BCD integers in registers rs1 and rs2.
+
+    Note: The existing CVA6 example supports only register-to-register instructions with up to three
+    source registers and a single destination register. New memory operations will need substantial modifications
+    to the coprocessor and CVA6 system-on-chip.
+
+#. Branch CVA6 repo
+
+    .. code-block::
+
+        git branch new_coprocessor
+        git checkout new_coprocessor
+
+#. Specialise the decoder function in core/cvxif_example/include/cvxif_instr_pkg.sv
+
+    Example new lines in cvxif_instr_pkg:
+
+    .. code-block::
+
+        // 1 RISCV instruction class for our Coprocessor
+        parameter int unsigned NbInstr = 1;
+        parameter copro_issue_resp_t CoproInstr[NbInstr] = '{
+          '{
+            instr: 32'b0000000_00000_00000_000_00000_0101011, // custom1 opcode
+            mask:  32'b1111111_00000_00000_110_00000_1111111, // bits that must match
+            resp : '{
+              accept : 1'b1,    // We will process this instruction
+              writeback : 1'b1, // We will write a general-purpose register
+              dualwrite : 1'b0,
+              dualread : 1'b0,
+              loadstore : 1'b0, // No memory operation  (Not supported in pipeline)
+              exc : 1'b0        // Cannot throw exception
+            }
+          }
+        };
```

Comment:
This looks based on a deprecated version of the `cvxif_instr_pkg.sv`

Reply:
I must have been working from the latest tagged release.  (v5.1.0 also reportedly doesn't build, sadly, due to a problem with problem with dual-issue parameterisation which was fixed soon after.). I'll update the example to be based on master.